### PR TITLE
Feature/EUI 6707

### PR DESCRIPTION
**Pull request by DavidJayakumar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-6707

### Change description ###

Advanced user search filter
Call to real api

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: api/staff-ref-data/real-api.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    res.status(status).send(groupOptions);
+  } catch (error) {
+    next(error);
+  }
+}
+
+export async function getFilteredUsers(req, res: Response, next: NextFunction) {
+  const { serviceCode, locations, skill, role, userType, jobTitle } = req.query;
+  console.log('req.query ', req.query);
```
Comment (udaydenduluri33) [inline api/staff-ref-data/real-api.ts:84]: Can you use this library ?
https://www.npmjs.com/package/query-string
This is a smarter way to deal with such things

Comment (udaydenduluri33) [inline api/staff-ref-data/real-api.ts:84]: Remove this console.log

File: api/staff-ref-data/real-api.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    });
+
+    res.status(status).send(groupOptions);
+  } catch (error) {
+    next(error);
+  }
+}
+
+export async function getFilteredUsers(req, res: Response, next: NextFunction) {
+  const queryString = require('query-string');
```
Comment (udaydenduluri33) [inline api/staff-ref-data/real-api.ts:83]: Please use import rather than ES6 require. lol

**Review by udaydenduluri33 — APPROVED** [review]

**Review by MateiHMCTS — APPROVED** [review]

File: api/staff-ref-data/real-api.ts (hunk 1 above)
Comment (DavidJayakumar) [reply api/staff-ref-data/real-api.ts]: done

**Pull request merged**